test/tools/OpenCover/OpenCover.psm1

```diff
@@ -524,15 +518,32 @@ function Invoke-OpenCover
                 if(-not $openCoverProcess)
                 {
                     #run must have completed.
+                    $openCoverExited = $true
                     break
                 }
             }
+
+            if(-not $openCoverExited)
+            {
+                throw "Opencover has not exited in 6 hours"
+            }
         }
         finally
         {
-            # set it back
-            $env:PSModulePath = $saveModPath
             Remove-Item "$env:temp\unelevated.ps1" -force -ErrorAction SilentlyContinue
         }
     }
 }
+
+function CreateOpenCoverCmdline($target, $outputLog, $targetArgs)
+{
+    $targetArgString = $targetArgs -join " "
+
+    $bytes = [System.Text.Encoding]::Unicode.GetBytes($targetArgString)
+    $base64targetArgs = [convert]::ToBase64String($bytes)
+
+    # the order seems to be important. Always keep -targetargs as the last parameter.
+    $cmdline = "-target:$target","-register:user","-output:${outputLog}","-nodefaultfilters","-oldstyle","-hideskipped:all","-mergeoutput", "-filter:`"+[*]* -[Microsoft.PowerShell.PSReadLine]*`"", "-targetargs:`"-EncodedCommand $base64targetArgs`""
```

Comment:
having these all on one line is a difficult to read, this would work as well:
```PowerShell
$cmdline = "-target:$target",
    "-register:user",
    "-output:${outputLog}",
    "-nodefaultfilters",
    "-oldstyle",
    "-hideskipped:all",
    "-mergeoutput",
    "-filter:`"+[*]* -[Microsoft.PowerShell.PSReadLine]*`"",
    "-targetargs:`"-EncodedCommand $base64targetArgs`""
```